src/apps/Permissions/AppRoles.js

```diff
@@ -72,6 +72,11 @@ class AppRoles extends React.PureComponent {
 }
 
 class RoleRow extends React.Component {
+  static propTypes = {
+    onManage: PropTypes.func.isRequired,
+    role: PropTypes.shape({ bytes: PropTypes.string }).isRequired,
+    manager: PropTypes.shape({ type: PropTypes.string }).isRequired,
```

Comment:
Not sure if it's worth being explicit here, as an inner component, but there's a few keys missing:

- `manager.address`, `manager.app`